.github/workflows/build.yml

```diff
@@ -0,0 +1,255 @@
+name: Build & Test
+
+on:
+  pull_request
+
+jobs:
+  unit-functional:
+    name: ${{ matrix.os }} / ${{ matrix.python }} / unit-functional
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
+        python:
+          - "3.7"
+          - "3.8"
+          - "3.9"
```

Comment:
nit: shall we add "3.10" as well?

Reply:
I think this might result in exceeding the limit of jobs/pr... Might need a separate job that does a smoke test with all versions.